src/webgl-surface/drawing/shape/curved-line-shape.ts

```diff
@@ -25,6 +25,29 @@ export interface IMarchingAnts {
   gapLength: number;
 }
 
+/**
+ * This gets the radian of line from center to point
+ *
+ * @param point
+ * @param center
+ */
+function getAngle(point: IPoint, center: IPoint): number {
```

Comment:
This looks like a cool chunk of math but my gut feels like something is awry here.

Generally if you're getting radian angles you should just use Math.atan2(y, x). That would at least reduce the costly sqrt and potential acos operation.